### PR TITLE
Add linter to Canary and fix go 1.23 linting issues

### PR DESCRIPTION
**Pull request by apostasie**
This adds a lint validation to the Canary target.
Currently done on windows (because of the go install on the host), but will move to an independent stage after cleaning up the go_version detection routine.

Once green, we should merge ASAP to help with #3320

### CODE REVIEW

File: .github/workflows/test-canary.yml [hunk 1]
```diff
@@ -92,6 +99,11 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
           cache: true
           check-latest: true
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6.1.0
+        with:
+          version: v1.59.1
```
Comment (AkihiroSuda) [inline .github/workflows/test-canary.yml:105]: For canary probably this should be left blank?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -222,7 +222,7 @@ jobs:
           if echo "${ROOTLESSKIT_VERSION}" | grep -q v1; then
             WORKAROUND_ISSUE_622=1
           fi
-          echo "WORKAROUND_ISSUE_622=${WORKAROUND_ISSUE_622}" >>$GITHUB_ENV
+          echo "WORKAROUND_ISSUE_622=${WORKAROUND_ISSUE_622}" >> "$GITHUB_ENV"
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:225]: ?

**Comment by AkihiroSuda** [comment]
Thanks, in the commit message could you mention the fixed errors such as `SA4032: due to the file's build constraints, runtime.GOOS will never equal "windows" (staticcheck)`?

The commits can be also split for the each of the SA%d fixes.

**Comment by AkihiroSuda** [comment]
> go 1.23 linting issues

IIUC this is the matter of golangci-lint rather than go itself?

File: .github/workflows/test.yml (hunk 2 above)
Comment (apostasie) [reply .github/workflows/test.yml:225]: That must be quoted to prevent splitting.

**Comment by apostasie** [comment]
> > go 1.23 linting issues
> 
> IIUC this is the matter of golangci-lint rather than go itself?

I think it is a combination of both? (eg: golangci-lint with that version of go)

Anyhow, thanks for the comments.

Will integrate all and split in different commits.

Right now struggling with windows (as usual), will re-request review when done.

**Comment by apostasie** [comment]
@AkihiroSuda canary lint is green. All comments addressed.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**